### ISSUE TITLE
[a11y] Toast not clearly visible in High Contrast

### ISSUE THREAD

**Issue by dpersing** [Shopify/polaris-react#1216]
# Issue summary

The toast is difficult to see against the background if it's viewed in Windows High Contrast, since the background color is stripped out. Adding a border (either a transparent one using `border: 1px solid transparent`, or one that's only displayed using the `ms-high-contrast` query. @AndrewMusgrave has done some work with high contrast styles, so you can likely ping either of us about this.

<img width="1314" alt="Toast in High Contrast" src="https://user-images.githubusercontent.com/1462085/53360891-9a608e80-38eb-11e9-96e4-9c9439ae35ec.png">

This issue was originally reported as part of a review of https://github.com/Shopify/polaris-react/pull/919, but I've opened this issue separately since it wasn't addressed there.

## Expected behavior

The toast should be as easy to discern in High Contrast as it is normally.

## Actual behavior

The toast blends into the page background.

## Steps to reproduce the problem

1. Open any toast example using Microsoft Edge (such as https://polaris.shopify.com/components/feedback-indicators/toast) in a Windows VM or Windows machine
1. Turn on a Windows High Contrast theme and refresh the page.
1. Activate the control to display the toast.

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 3.10.0+
- Browser: Edge (and other Windows browsers)
- Device: Windows VM on Mac
- Operating System: Windows 10
